### PR TITLE
Hotfix/civ 1683 redis data is cleared by your details screen

### PR DESCRIPTION
**Pull request by LeonardoPalmeiro**
### JIRA link ###

CIV-1683

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/modules/citizenDetails/citizenDetailsService.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+export const saveRespondent = async(claimId: string, citizenAddress: CitizenAddress, citizenCorrespondenceAddress: CitizenCorrespondenceAddress) : Promise<void> => {
+  const responseData = await getCaseDataFromStore(claimId) || new Claim();
+  if (get(responseData, 'respondent1.primaryAddress')) {
+    responseData.respondent1.primaryAddress = buildPrimaryAddress(citizenAddress);
+  }
+  if (get(responseData, 'respondent1.correspondenceAddress')) {
+    responseData.respondent1.correspondenceAddress = buildCorrespondenceAddress(citizenCorrespondenceAddress);
+  }
+  if (!(get(responseData, 'respondent1.primaryAddress') && get(responseData, 'respondent1.correspondenceAddress'))) {
+    const respondent = new Respondent();
```
Comment (osheaf01) [inline src/main/modules/citizenDetails/citizenDetailsService.ts:31]: Did you mean !get primaryAddress && !get correspondenceAddress here?

File: src/main/routes/features/response/citizenDetails/citizenDetailsController.ts [hunk 2]
```diff
@@ -1,156 +1,112 @@
...
-            claim.respondent1.correspondenceAddress.PostCode);
-        } else {
-          formCorrespondenceModel = new CitizenCorrespondenceAddress();
-        }
-      } else {
-        claim = new Claim();
-      }
+    const responseDataRedis: Respondent = await getRespondentInformation(req.params.id);
+    if (!_.isEmpty(responseDataRedis)) {
+      formAddressModel = new CitizenAddress(
```
Comment (osheaf01) [inline src/main/routes/features/response/citizenDetails/citizenDetailsController.ts:44]: Don't like "formAddressModel" - citizenAddressModel?

**Review by osheaf01 — APPROVED** [review]
Overall, LGTM, just a couple of inline comments/queries.

File: src/main/routes/features/response/citizenDetails/citizenDetailsController.ts [hunk 3]
```diff
@@ -1,156 +1,112 @@
...
+    const errorList = new Form();
+    if (req.body.postToThisAddress === YesNo.YES) {
+      citizenAddress.errors = validator.validateSync(citizenAddress);
+      citizenCorrespondenceAddress.errors = validator.validateSync(citizenCorrespondenceAddress);
+      errorList.errors = citizenAddress.errors.concat(citizenCorrespondenceAddress.errors);
+    } else {
+      citizenAddress.errors = validator.validateSync(citizenAddress);
+      errorList.errors = citizenAddress.errors;
+    }
+    if ((citizenAddress.errors && citizenAddress.errors.length > 0)
```
[the remark this answers is not in the document]
Comment (LeonardoPalmeiro) [reply src/main/routes/features/response/citizenDetails/citizenDetailsController.ts:100]: I didn't touch in that line, was there in the preview code.
But I can change.

File: src/main/modules/citizenDetails/citizenDetailsService.ts (hunk 1 above)
Comment (LeonardoPalmeiro) [reply src/main/modules/citizenDetails/citizenDetailsService.ts:31]: in this case I want to test both in the negative condition together 
but I believe we could use something like this. !(get(responseData, ['respondent1.primaryAddress', 'respondent1.correspondenceAddress']).
I will have a look.

Thanks.

File: src/main/routes/features/response/citizenDetails/citizenDetailsController.ts [hunk 4]
```diff
@@ -1,156 +1,112 @@
...
+      || (citizenCorrespondenceAddress.errors && citizenCorrespondenceAddress.errors.length > 0)) {
+      renderPageWithError(res, citizenAddress, citizenCorrespondenceAddress, errorList, req);
```
[the remark this answers is not in the document]
Comment (LeonardoPalmeiro) [reply src/main/routes/features/response/citizenDetails/citizenDetailsController.ts:102]: I didn't touch in this logic.

File: src/main/modules/citizenDetails/citizenDetailsService.ts (hunk 1 above)
Comment (LeonardoPalmeiro) [reply src/main/modules/citizenDetails/citizenDetailsService.ts:31]: done

File: src/main/routes/features/response/citizenDetails/citizenDetailsController.ts (hunk 2 above)
Comment (LeonardoPalmeiro) [reply src/main/routes/features/response/citizenDetails/citizenDetailsController.ts:44]: I didn't touch in this line, but I could change.
I changed to citizenAddressModel and citizenCorrespondenceAddressModel
it's done

File: src/main/routes/features/response/citizenDetails/citizenDetailsController.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (LeonardoPalmeiro) [reply src/main/routes/features/response/citizenDetails/citizenDetailsController.ts:100]: was we talked in this case I created a new issue https://tools.hmcts.net/jira/browse/CIV-1749

File: src/main/routes/features/response/citizenDetails/citizenDetailsController.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (LeonardoPalmeiro) [reply src/main/routes/features/response/citizenDetails/citizenDetailsController.ts:102]: yes in this case in the screen has two section and should to do the validation separately

**Pull request merged**